### PR TITLE
feat: create file resource while deploying service pack for viya

### PR DESCRIPTION
**Pull request by saadjutt01**
## Issue

CLI issue: https://github.com/sasjs/cli/issues/707

## Intent

While deploying service-pack for viya, should upload files with file api

## Implementation

Added `createFile` for viya.

## Checks

- [x] Code is formatted correctly (`npm run lint:fix`).
- [ ] All unit tests are passing (`npm test`).
- [ ] All `sasjs-cli` unit tests are passing (`npm test`).
- [ ] All `sasjs-tests` are passing (instructions available [here](https://github.com/sasjs/adapter/blob/master/sasjs-tests/README.md)).

### CODE REVIEW

File: src/SASViyaApiClient.ts [hunk 1]
```diff
@@ -536,6 +538,53 @@ export class SASViyaApiClient {
       .then((res) => res.result)
   }
 
+  /**
+   * Creates a file. Path to or URI of the parent folder is required.
+   * @param fileName - the name of the new file.
+   * @param content - the content of the new file.
```
Comment (allanbowe) [inline src/SASViyaApiClient.ts:544]: should this be `contentBuffer` ?

**Review by allanbowe — CHANGES_REQUESTED** [review]

File: src/SASViyaApiClient.ts (hunk 1 above)
Comment (saadjutt01) [reply src/SASViyaApiClient.ts:544]: fixed

File: src/SASjs.ts [hunk 2]
```diff
@@ -300,6 +300,41 @@ export default class SASjs {
     )
   }
 
+  /**
+   * Creates a file at SAS file system.
```
[the remark this answers is not in the document]
Comment (saadjutt01) [reply src/SASjs.ts:304]: updated, and also for createFolder's TSDoc comments

File: src/SASjs.ts [hunk 3]
```diff
@@ -300,6 +300,41 @@ export default class SASjs {
...
+   * @param fileName - name of the file to be created.
+   * @param content - content of the file to be created.
+   * @param parentFolderPath - the full path (eg `/Public/example/myFolder`) of the parent folder.
+   * @param parentFolderUri - the URI of the parent folder.
+   * @param accessToken - the access token to authorizing the request.
+   * @param sasApiClient - a client for interfacing with SAS API.
+   * @param isForced - flag that indicates if target folder already exists, it and all subfolders have to be deleted. Applicable for SAS VIYA only.
```
[the remark this answers is not in the document]
Comment (saadjutt01) [reply src/SASjs.ts:311]: removed

**Review by allanbowe — APPROVED** [review]

**Pull request merged**